lib/chart/internal/utils.ex

```diff
@@ -1,15 +1,12 @@
 defmodule Chart.Internal.Utils do
   @moduledoc false
 
-  def put(module, key, config_key, config, validate_fun) do
-    guard = validate_fun.() |> Map.fetch!(key)
+  alias ExMaybe, as: Maybe
 
-    validated_val =
-      config
-      |> Keyword.get(config_key, Map.fetch!(module, key))
-      |> guard.()
-
-    Map.put(module, key, validated_val)
+  def update_module(settings, kw, validate) do
```

Comment:
Ten nazev moc neodpovida. Plni to strukturu daty nebo lepe je to merge mezi strukturou a daty/nastavenim od uzivatele.